### DISCUSSION TITLE
What are artifacts files to be created in "GraphRoot"/overlay-containers/"containerID"/userdata/artifacts?

### DISCUSSION THREAD

**Discussion by ttsuuubasa** [containers/podman#16368] [Q&A] [answered]
I'm not sure what is for the directory "GraphRoot"/overlay-containers/"containerID"/userdata/artifacts
like /var/lib/containers/storage/overlay-containers/"containerID"/userdata/artifacts.

This "artifacts" subdirectory is created at libpod.setupStorage() (*1).

Although libpod.AddArtifact() (*2) is supposed to create the "artifacts" files in that directory,
it looks like this function is not called from anywhere.

And there is a process in "podman container checkpoint --export" to export this directory as includeFiles (*3).

Previously, libpod (e.g. podman v1.4.2-stable) create a file "create-config" for pre OCI spec as the "artifacts" files (*4).
For now, I think this directory may have no use because the process to create "artifacts" files is not found.

What is the purpose of "artifacts" files?

(*1) https://github.com/containers/podman/blob/main/libpod/container_internal.go#L543
(*2) https://github.com/containers/podman/blob/main/libpod/container_api.go#L484
(*3) https://github.com/containers/podman/blob/main/libpod/container_internal_common.go#L856
(*4) https://github.com/containers/podman/blob/v1.4.2-stable/cmd/podman/shared/create.go#L738

**Comment by mheon** [comment, marked as answer]
Artifacts are intended to store metadata files associated with a container. In the past, this included information used for `podman inspect`, but we refactored to not require that a long time ago. As of now, the functionality is unused, but available if we ever need such storage in the future.
